File: Accommodation-Reservation-Backend/src/main/java/com/accommodationsite/accommodationreservationapp/model/Accommodation.java
package com.accommodationsite.accommodationreservationapp.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Accommodation implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    @Column(nullable = false)
    private String name;
    private String status;
    private String mainPagePicture;
    private String secondImage;
    private String thirdImage;
    private String description;
    private String checkInDescriptionForEmail;
    private String city;

    @OneToOne(targetEntity = Address.class, cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private Address address;

    @OneToOne(targetEntity = Person.class, cascade = CascadeType.MERGE)
    @JoinColumn(name = "person_id", referencedColumnName = "id")
    private Person person;

    @ManyToMany(targetEntity = Amenity.class, cascade = CascadeType.ALL)
    @JoinColumn(name = "amenities",  referencedColumnName = "id")
    private List<Amenity> amenities;


    private String phoneNumber;
}

File: Accommodation-Reservation-Frontend/src/app/shared/models/accommodation.ts
import { Address } from "./address";
import { Amenity } from "./amenity";
import { Person } from "./person";
import { Room } from "./room";

export interface Accommodation {
    id?: number;
    name: string;
    status?: string;
    phoneNumber?: string;
    mainPagePicture?: string;
    secondImage?: string;
    thirdImage?: string;
    description?: string;
    checkInDescriptionForEmail?: string;
    city: string;
    address?: Address;
    rooms?: Room[];
    amenities?: Amenity[];
    person?: Person;
}